resources/blog/posts/2023/2023-08-21-using-scoped-traverse.md

```diff
@@ -0,0 +1,84 @@
+---
+id: 51
+title: "Using Scoped Traverse to Locate Specific Node"
+perex: |
+    Rector no longer support parent lookup since version 0.17, so lookup specific node will need to resolve from parent to child instead. You can traverse with `SimpleCallableNodeTraverser`.
+---
+
+For example, on the following code:
+
+```php
+class SomeClass
+{
+    public function run()
+    {
+        $someClosure = function () {
+            return 1;
+        };
+
+        return $someClosure() + 2;
+    }
+}
+```
+
+You want to collect return values, but not inside inner scope (_anonymous class_, _inner function_, or _closure_), you can use `PhpParser\Node\Stmt\ClassMethod` as node types:
+
+```php
+use PhpParser\Node\Stmt\ClassMethod;
+
+// ...
+public function getNodeTypes()
+{
+    return [ClassMethod::class];
+}
+```
+
+Then, you can do in `refactor()` method:
+
+```php
+use PhpParser\Node\Expr\Closure;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\Function_;
+use PhpParser\NodeTraverser;
+
+public function refactor(Node $node): ?Node
+{
+    if ($node->stmts === null) {
+        return null;
+    }
+
+    // initialize collected returns
+    $returns = [];
+    $this->traverseNodesWithCallable(
```

Comment:
There is `findInstancesOfInFunctionLikeScoped()` already for only this use case, so it need more specific use case

Reply:
I updated to more specific use case https://github.com/rectorphp/getrector-com/pull/1587/commits/4f6a129f28047050d81569ef06d2a8d4cf22ac0d